### PR TITLE
(Fix) Calculate real min cap to the moment of invest

### LINKED ISSUE #853: (Wizard 2.0) Disable to buy minimal amount(decimals) if decimals>16
Test URL: https://5aec92f21f12b70339c64d4a--architect-coin-64174.netlify.com/

Steps:
1. Create crowdsale with any valid parameters, decimals =17 or 18. 
2. Open invest page
3. Try to buy minimal amount: 0.000000000000000001 for decimals =18

Expected result:
 - user should be able to buy any amount from diapason : decimals ... total supply
Actual result: 
 - disabled to buy decimals
- loader displayed, transaction isn't created

If you are reporting a problem with Token Wizard, please include the following information:

### Which network did you use? (Mainnet, Kovan, Rinkeby, etc.)

*Ropsten*

### If you were able to create it, what is the URL of your crowdsale?

*https://5aec92f21f12b70339c64d4a--architect-coin-64174.netlify.com/invest?exec-id=0x45cd5779ef7840269cf3fc9e2490ea72ff119027401f8ac0a61c932cd1b64c5f&networkID=3
*

### Do you have screenshots showing the problem?

screen shot 2018-05-06 at 11 03 38

### Do you see errors in the dev console? If yes, please include a screenshot

To open the dev console in Google Chrome, press F12, or go to `View -> Developer -> Developer Tools`, and open the Console tab

If you see errors, please right click on them and "Save as..". Zip saved file and attach it to the Issue.

*Your answer*

---

### PR DESCRIPTION
**Pull request by fernandomg**
Closes #853 

I created `getUserMinLimits` which contemplates both strategies and with/without whitelists enabled.

I added new field in the invest page, to proactively inform the user what's the minimum acceptable value. Also used that value to validate the `input` field.

### CODE REVIEW

**Comment by fernandomg** [comment]
> For Dutch Auction with global min cap the introduced field displays natural min cap for me.

@vbaranov can you try with this execID in ropsten? `0x711336caa77e91c381cec45202db81803a987f02d6da7e6471b0c0649303c76b`

I set a `minCap = 23` and it displayed `23` for min contribution, then I bought `23` tokens and now I get the natural min cap.

File: src/components/invest/index.js [hunk 1]
```diff
@@ -439,6 +459,10 @@ export class Invest extends React.Component {
               <p className="hashes-title">{totalSupply} {tokenTicker}</p>
               <p className="hashes-description">Total Supply</p>
             </div>
+            <div className="hashes-i">
+              <p className="hashes-title">{minimumContribution}</p>
+              <p className="hashes-description">Minimum Contribution</p>
```
[the remark this answers is not in the document]
Comment (fernandomg) [reply src/components/invest/index.js]: fixed, added `tokenTicker` to indicate the unit.

**Comment by vbaranov** [comment]
Ah, this is because token's total supply > crowdsale's hard cap. The difference is on the owner's account. And I open the page with the owner of crowdsale.

**Comment by vbaranov** [comment]
Thus, it is not so critical. I'm gonna to merge.

**Pull request merged**